### PR TITLE
Filter locked users in the admin API

### PR DESCRIPTION
**Pull request by hanadi92**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: [Hanadi](https://github.com/hanadi92)
This PR is related to this issue (feature request): https://github.com/matrix-org/synapse/issues/16323

### CODE REVIEW

**Review by DMRobertson — DISMISSED** [review]
Seems reasonable, many thanks for this. Can you also update the [admin API docs](https://matrix-org.github.io/synapse/latest/admin_api/user_admin_api.html#list-accounts) to

- describe the new `locked` query parameter,
- explaining that the default behaviour is to exclude locked users, and
- note which version of Synapse this parameter was added in.

**Comment by hanadi92** [comment]
Sure thing!

>     * note which version of Synapse this parameter was added in.
Could you please help me out with the version? Would that be v1.92.2 or upcoming v1.92.3 or v1.93?

**Comment by DMRobertson** [comment]
> Could you please help me out with the version? Would that be v1.92.2 or upcoming v1.92.3 or v1.93?

Let's go with1.93---we expect to cut that release tomorrow. (The team can always bump the version string in the docs if it doesn't land in time.)

File: changelog.d/16328.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Enables filtering locked users in list users admin API.
```
Comment (DMRobertson) [inline changelog.d/16328.feature:1]:
```suggestion
Report whether a user is `locked` in the [List Accounts admin API](https://matrix-org.github.io/synapse/latest/admin_api/user_admin_api.html#list-accounts), and exclude locked users by default.
```

File: docs/admin_api/user_admin_api.md [hunk 2]
```diff
@@ -274,7 +280,7 @@ The following fields are returned in the JSON response body:
   - `avatar_url` - string -  The user's avatar URL if they have set one.
   - `creation_ts` - integer - The user's creation timestamp in ms.
   - `last_seen_ts` - integer - The user's last activity timestamp in ms.
-
+  - `locked` - bool - Status if that user has been marked as locked.
 - `next_token`: string representing a positive integer - Indication for pagination. See above.
 - `total` - integer - Total number of media.
```
Comment (DMRobertson) [inline docs/admin_api/user_admin_api.md:286]:
```suggestion

*Added in Synapse 1.93:* the `locked` query parameter and response field.

File: docs/admin_api/user_admin_api.md [hunk 3]
```diff
@@ -274,7 +280,7 @@ The following fields are returned in the JSON response body:
   - `avatar_url` - string -  The user's avatar URL if they have set one.
   - `creation_ts` - integer - The user's creation timestamp in ms.
   - `last_seen_ts` - integer - The user's last activity timestamp in ms.
-
+  - `locked` - bool - Status if that user has been marked as locked.
```
Comment (DMRobertson) [inline docs/admin_api/user_admin_api.md:283]:
```suggestion
  - `locked` - bool - Status if that user has been marked as locked. Note: Introduced in v1.93.
```

**Review by DMRobertson — COMMENTED** [review]
Thanks, just a few wording tweaks.

**Review by MatMaul — APPROVED** [review]
LGTM.

**Pull request merged**